api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -291,6 +290,147 @@ public void testInvitationKeyVerification_invitationKeyMismatch() {
+            .duaTypeEnum(DuaType.RESTRICTED);
+    institutionService.createInstitution(broad);
+
+    final VerifiedInstitutionalAffiliation verifiedInstitutionalAffiliation =
+        new VerifiedInstitutionalAffiliation()
+            .institutionShortName("Broad")
+            .institutionalRoleEnum(InstitutionalRole.STUDENT);
+    createAccountRequest
+        .getProfile()
+        .verifiedInstitutionalAffiliation(verifiedInstitutionalAffiliation)
+        .contactEmail("notBob@example.com");
+    profileController.createAccount(createAccountRequest);
+  }
+
+  @Test(expected = BadRequestException.class)
+  public void testCreateAccount_MismatchEmailDomain() {
+    createUser();
+    config.featureFlags.requireInstitutionalVerification = true;
+    final Institution broad =
+        new Institution()
+            .shortName("Broad")
+            .displayName("The Broad Institute")
+            .emailAddresses(Collections.singletonList(CONTACT_EMAIL))
+            .emailDomains(Collections.singletonList("example.com"))
+            .duaTypeEnum(DuaType.MASTER);
+    institutionService.createInstitution(broad);
+
+    final VerifiedInstitutionalAffiliation verifiedInstitutionalAffiliation =
+        new VerifiedInstitutionalAffiliation()
+            .institutionShortName("Broad")
+            .institutionalRoleEnum(InstitutionalRole.STUDENT);
+    createAccountRequest
+        .getProfile()
+        .verifiedInstitutionalAffiliation(verifiedInstitutionalAffiliation)
+        .contactEmail("bob@broad.com");
+    profileController.createAccount(createAccountRequest);
+  }
+
+  @Test(expected = BadRequestException.class)
+  public void testCreateAccount_MismatchEmailDomainNullDUA() {
+    createUser();
+    config.featureFlags.requireInstitutionalVerification = true;
+    final Institution broad =
+        new Institution()
+            .shortName("Broad")
+            .displayName("The Broad Institute")
+            .emailAddresses(Collections.singletonList(CONTACT_EMAIL))
+            .emailDomains(Collections.singletonList("example.com"));
+    institutionService.createInstitution(broad);
+
+    final VerifiedInstitutionalAffiliation verifiedInstitutionalAffiliation =
+        new VerifiedInstitutionalAffiliation()
+            .institutionShortName("Broad")
+            .institutionalRoleEnum(InstitutionalRole.STUDENT);
+    createAccountRequest
+        .getProfile()
+        .verifiedInstitutionalAffiliation(verifiedInstitutionalAffiliation)
+        .contactEmail("bob@broadInstitute.com");
+    profileController.createAccount(createAccountRequest);
+  }
+
+  @Test
+  public void testCreateAccount_Success_RESTRICTEDDUA() {
+    createUser();
+    config.featureFlags.requireInstitutionalVerification = true;
+    final Institution broad =
+        new Institution()
+            .shortName("Broad")
+            .displayName("The Broad Institute")
+            .emailAddresses(Collections.singletonList("bob@example.com"))
```

Comment:
please make this like use `CONTACT_EMAIL` and change the domain to a different domain